scarlet/initialization.py

```diff
@@ -0,0 +1,438 @@
+    if mask.sum() > 0:
+        morph[~mask] = 0
+
+        # normalize to unity at peak pixel
+        center_morph = morph[pixel_center[0], pixel_center[1]]
+        morph /= center_morph
+
+        # find fitting bbox
+        bbox = Box.from_data(morph, min_value=0)
+        if bbox.contains(pixel_center):
+            size = 2 * max(
+                (
+                    pixel_center[0] - bbox.start[-2],
+                    bbox.stop[0] - pixel_center[-2],
+                    pixel_center[1] - bbox.start[-1],
+                    bbox.stop[1] - pixel_center[-1],
+                )
+            )
+            while boxsize < size:
+                boxsize *= 2
+    else:
+        msg = "No flux above threshold for source at y={0} x={1}".format(*pixel_center)
+        logger.warning(msg)
+
+    # define bbox and trim to bbox
+    bottom = pixel_center[0] - boxsize // 2
+    top = pixel_center[0] + boxsize // 2
+    left = pixel_center[1] - boxsize // 2
+    right = pixel_center[1] + boxsize // 2
+    bbox = Box.from_bounds((bottom, top), (left, right))
+    morph = bbox.extract_from(morph)
+    bbox_3d = Box.from_bounds((0, frame.C), (bottom, top), (left, right))
+    return morph, bbox_3d
+
+
+def init_extended_source(
+    sky_coord,
+    frame,
+    observations,
+    coadd=None,
+    bg_cutoff=None,
+    thresh=1,
+    symmetric=True,
+    monotonic=True,
+    min_grad=0.2,
+):
+    """Initialize the source that is symmetric and monotonic
+    See `ExtendedSource` for a description of the parameters
+    """
+    try:
+        iter(observations)
+    except TypeError:
+        observations = [observations]
+    # determine initial SED from peak position
+    # SED in the frame for source detection
+    seds = []
+    for obs in observations:
+        _sed = get_psf_sed(sky_coord, obs, frame)
+        seds.append(_sed)
+
+    sed = np.concatenate(seds).flatten()
+    if np.all(sed <= 0):
+        # If the flux in all channels is  <=0,
+        msg = f"Zero or negative SED {sed} at y={sky_coord[0]}, x={sky_coord[1]}"
+        logger.warning(msg)
+    if coadd is None:
+        # which observation to use for detection and morphology
+        try:
+            bg_rms = np.array([[1 / np.sqrt(w[w > 0].mean()) for w in obs_.weights] for obs_ in observations])
+        except:
+            raise AttributeError(
+                "Observation.weights missing! Please set inverse variance weights"
+            )
+        coadd, bg_cutoff = build_sed_coadd(seds, bg_rms, observations)
+    else:
+        if bg_cutoff is None:
+            raise AttributeError(
+                "background cutoff missing! Please set argument bg_cutoff"
+            )
+    # Apply the necessary constraints
+    center = frame.get_pixel(sky_coord)
+    if symmetric:
+        morph = operator.prox_uncentered_symmetry(
+            coadd*1, 0, center=center, algorithm="sdss" # *1 is to artificially pass a variable that is not coadd
```

Comment:
why `coadd*1`? Why is the multiplication needed?

Reply:
Replaced by .copy()